wazo_dird/plugins/csv_backend/plugin.py

```diff
@@ -94,16 +100,27 @@ def _load_file(self):
 
         filename = self._config['file']
         delimiter = str(self._config.get('separator', ','))
-
+        fname = pathlib.Path(filename)
         try:
-            logger.debug('Reading %s with delimiter %r', filename, delimiter)
-            with open(filename, 'r') as f:
-                csvreader = csv.reader(f, delimiter=delimiter)
-                keys = [key for key in next(csvreader)]
-                self._content = [self._row_to_dict(keys, row) for row in csvreader]
-                logger.debug('Loaded with %s', self._content)
-        except IOError:
-            logger.exception('Could not load CSV file content')
+            tmp_csv_file_last_modification_date = fname.stat().st_mtime
+            if self._csv_last_modification_time != tmp_csv_file_last_modification_date:
+                try:
+                    logger.debug('Reading %s with delimiter %r', filename, delimiter)
+                    with open(filename, 'r') as f:
+                        csvreader = csv.reader(f, delimiter=delimiter)
+                        keys = [key for key in next(csvreader)]
+                        self._content = [
+                            self._row_to_dict(keys, row) for row in csvreader
+                        ]
+                        logger.debug('Loaded with %s', self._content)
+                except IOError:
+                    logger.exception('Could not load CSV file content')
+                finally:
+                    self._csv_last_modification_time = (
```

Comment:
I think this line needs to be in the `try` block. If an `IOError` we do not want to save the last modification time since we failed to read the file.